code/modules/xenoarchaeology/traits/activators.dm

```diff
@@ -0,0 +1,671 @@
+	blacklist_traits = list(/datum/xenoartifact_trait/minor/dense)
+	weight = 32
+	incompatabilities = TRAIT_INCOMPATIBLE_MOB | TRAIT_INCOMPATIBLE_STRUCTURE
+
+/datum/xenoartifact_trait/activator/weighted/New(atom/_parent)
+	. = ..()
+	if(!parent?.parent)
+		return
+	RegisterSignal(parent?.parent, COMSIG_ITEM_EQUIPPED, TYPE_PROC_REF(/datum/xenoartifact_trait/activator, translation_type_d))
+
+/datum/xenoartifact_trait/activator/weighted/translation_type_d(datum/source, atom/target)
+	trigger_artifact(target, XENOA_ACTIVATION_TOUCH)
+
+/*
+	Pitched
+	This trait activates the artifact when it is thrown
+*/
+/datum/xenoartifact_trait/activator/pitched
+	label_name = "Pitched"
+	label_desc = "Pitched: The artifact seems to be made of an aerodynamic material. This material seems to be triggered by motion, such as being thrown."
+	flags = XENOA_BLUESPACE_TRAIT | XENOA_URANIUM_TRAIT | XENOA_BANANIUM_TRAIT | XENOA_PEARL_TRAIT
+	blacklist_traits = list(/datum/xenoartifact_trait/minor/dense)
+	weight = -8
+	incompatabilities = TRAIT_INCOMPATIBLE_MOB | TRAIT_INCOMPATIBLE_STRUCTURE
+
+/datum/xenoartifact_trait/activator/pitched/New(atom/_parent)
+	. = ..()
+	if(!parent?.parent)
+		return
+	RegisterSignal(parent?.parent, COMSIG_MOVABLE_IMPACT, TYPE_PROC_REF(/datum/xenoartifact_trait/activator, translation_type_a))
+
+/*
+	Hungry
+	This trait activates the artifact when it's fed
+*/
+/datum/xenoartifact_trait/activator/sturdy/hungry
+	material_desc = null
+	label_name = "Hungry"
+	label_desc = "Hungry: The artifact seems to be made of a semi-living, hungry, material. This material seems to be triggered by feeding interactions."
+	flags = XENOA_BLUESPACE_TRAIT | XENOA_URANIUM_TRAIT | XENOA_BANANIUM_TRAIT | XENOA_PEARL_TRAIT
+	weight = 32
+	///How much damage do we deal per bite?
+	var/eat_damage = 15
+	///Timer logic for biting people
+	var/bite_cooldown = 4 SECONDS
+	var/bite_timer
+	///Will we tollerate the taste of humans? Used for subtypes
+	var/maneater = FALSE
+
+/datum/xenoartifact_trait/activator/sturdy/hungry/trigger_artifact(atom/target, type, force)
+	if(parent.anti_check(target, type) || parent.calcified || get_dist(target, parent?.parent) > parent?.target_range)
+		return FALSE
+	//Find a food item
+	var/mob/living/M = target
+	var/edible
+	if(isliving(M))
+		var/list/sides = list("left", "right")
+		for(var/i in sides)
+			var/atom/A = M.get_held_items_for_side(i)
+			if(A) //Not pre-checking A can cause some runtimes
+				edible = SEND_SIGNAL(A, COMSIG_FOOD_FEED_ITEM, parent?.parent)
+	if(!edible && target)
+		edible = SEND_SIGNAL(target, COMSIG_FOOD_FEED_ITEM, parent?.parent)
+	//If food
+	var/atom/movable/AM = parent.parent
+	if(edible)
+		playsound(AM.loc, 'sound/items/eatfood.ogg', 60, 1, 1)
+		return ..()
+	//Otherwise, nibble the target, and spit them out, they're gross, ew
+	if(isliving(M) && !bite_timer)
+		playsound(AM.loc, 'sound/weapons/bite.ogg', 60, 1, 1)
+		AM.do_attack_animation(M)
+		M.adjustBruteLoss(eat_damage)
```

Comment:
This ignores all armour and applies to all parts of the target equally. It should also cause bleeding.

Reply:
I'll look into the armor part, but I think it might be too annoying if it caused bleeding